### PR TITLE
fix Adding/removing user from contacts duplicating their messages in …

### LINKED ISSUE #13892: Adding/removing user from contacts duplicating their messages in profile
# Bug Report

## Problem 
If adding/removing a user via "Add to contacts" button in the user profile, it duplicates messages (if any) in their status.
Messages are shown there until next rel-ogin.

#### Expected behavior
Messages are not changed

#### Actual behavior

https://user-images.githubusercontent.com/4557972/187448339-cde146c5-8687-40af-8833-d98b813dd716.mp4

### Reproduction

1) User1 and User2: adds each other to contact
2) User1: post smth to status
3) User1: open profile of User2 (chat > ...> View profile or via contacts)
 4) User2: add/remove/re-add User1 to contacts

### Additional Information

- Status version: RC1 1.20 
- Operating System: Android, iOS

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/13892

status: ready

### CODE REVIEW

**Comment by Parveshdhull** [comment]
Hi, this description is misleading as the visibility of this dialog is not dependent on the contact state, but on the availability of status updates. This means this dialog doesn't hide on adding contact but on receiving new messages.
So, I made changes that status updates should only be visible after contact is added and until then show this dialog. Please let me know if this is desired behavior or if I should improve or revert.

<div align="center">
Screenshot
</div>
cc: @churik 

**UPD:**
Thank you @flexsurfer for confirming desired behavior here. (We should also see messages for not added contacts)

(It's just that dialog showing, Add contact while already added as contact looked a little suspicious to me.
We can leave this for now, and maybe condition this message in next release, I guess.
And, as Status updates also shown in profile, better message might be
"Add the profile as a contact to receive updates **also** on your timeline")

File: src/status_im/ui/screens/profile/contact/views.cljs [hunk 1]
```diff
@@ -229,4 +228,4 @@
          :render-data               {:chat-id current-chat-id
                                      :profile true}
          :render-fn                 status.views/render-message
-         :data                      messages}]])))
+         :data                      (when (:added? contact) messages)}]])))
```
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/contact/views.cljs]: we should see messages for non added contacts

File: src/status_im/chat/models/message.cljs [hunk 2]
```diff
@@ -137,10 +137,11 @@
...
 
 (defn reduce-js-statuses [db ^js message-js]
-  (let [chat-id (.-localChatId message-js)
+  (let [chat-id             (.-localChatId message-js)
         profile-initialized (get-in db [:pagination-info chat-id :messages-initialized?])
-        timeline-message (timeline-message? db chat-id)]
-    (if (or profile-initialized timeline-message)
+        timeline-message    (timeline-message? db chat-id)
+        old-message         (get-in db [:messages chat-id (.-id message-js)])]
+    (if (and (or profile-initialized timeline-message) (nil? old-message))
```
Comment (flexsurfer) [inline src/status_im/chat/models/message.cljs:144]: could you elaborate ? im not sure how new messages will be added then ?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/profile/contact/views.cljs (hunk 1 above)
Comment (Parveshdhull) [reply src/status_im/ui/screens/profile/contact/views.cljs]: Hi @flexsurfer, Thank you for reviewing the PR and confirming desired behavior. (changes reverted)

File: src/status_im/chat/models/message.cljs (hunk 2 above)
Comment (Parveshdhull) [reply src/status_im/chat/models/message.cljs:144]: Hi, this flag only checks if we already have that message and skips that, otherwise just normal flow.

Comment (flexsurfer) [reply src/status_im/chat/models/message.cljs:144]: oh right, just misread it, thank you

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
This one can be merged and cherry-picked too.
Thanks @Parveshdhull!

**Pull request merged**